app/src/components/inputs/inputUserSearch/inputUserSearch.jsx

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2017 EPAM Systems
+ *
+ *
+ * This file is part of EPAM Report Portal.
+ * https://github.com/reportportal/service-ui
+ *
+ * Report Portal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Report Portal is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import Select from 'react-select';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { FormattedMessage } from 'react-intl';
+import { ScrollWrapper } from 'components/main/scrollWrapper/scrollWrapper';
+import { fetch, addTokenToImagePath, validate } from 'common/utils';
+import styles from './inputUserSearch.scss';
+import { UserItem } from './userItem';
+
+const cx = classNames.bind(styles);
+
+export const InputUserSearch = ({ isAdmin, projectId, onChange }) => {
+  const formURL = (input) => {
+    const apiVersion = '../api/v1/';
+    const page = 1;
+    const size = 10;
+    const sort = '&page.sort=login,ASC';
+    let search = '';
+    let startUrl;
+    if (!isAdmin) {
+      startUrl = `${apiVersion}project/${projectId}/usernames/`;
+    } else {
+      startUrl = `${apiVersion}user/`;
+    }
+    if (input) {
+      startUrl += 'search/';
+      search = `&term=${input}`;
+    } else {
+      startUrl += 'all';
+    }
+    startUrl += `?page.page=${page}&page.size=${size}${sort}${search}`;
+    return startUrl;
+  };
+  const getPhoto = (userId) => {
+    const d = new Date();
+    return addTokenToImagePath(`../api/v1/data/userphoto?v=${d.getTime()}&id=${userId}`);
+  };
+  const formOptions = (options) => {
+    const newOptions = options.map((option) => {
+      const tempOption = {};
+      tempOption.userName = option.full_name;
+      tempOption.userLogin = option.userId;
+      if (option.email) {
+        tempOption.email = option.email;
+      }
+      tempOption.disabled = false;
+      tempOption.isAssigned = false;
+      if (option.assigned_projects[projectId]) {
+        tempOption.isAssigned = true;
+        tempOption.disabled = true;
+      }
+      tempOption.userAvatar = getPhoto(option.userId);
+      return tempOption;
+    });
+    console.log(newOptions);
+    return newOptions;
+  };
+  const getUsers = (input) => {
```

Comment:
All async manipulations and data fetching & prepearing should be moved into ComponentDecorator (Container). 
Presentational compopnent should receive all preapeared data through the props from ComponentDecorator.